diff --git a/json_tokener.c b/json_tokener.c
index 04f11ba..a91fbc4 100644
--- a/json_tokener.c
+++ b/json_tokener.c
@@ -195,7 +195,7 @@ struct json_object* json_tokener_parse_ex(struct json_tokener *tok,
 
     case json_tokener_state_eatws:
       /* Advance until we change state */
-      while (isspace(c)) {
+      while (isspace((int)c)) {
 	if ((!ADVANCE_CHAR(str, tok)) || (!POP_CHAR(c, tok)))
 	  goto out;
       }
