diff --git a/config.sub b/config.sub
index 2a55a50..7a70afb 100755
--- a/config.sub
+++ b/config.sub
@@ -230,6 +230,10 @@ case $os in
 	-windowsnt*)
 		os=`echo $os | sed -e 's/windowsnt/winnt/'`
 		;;
+        -gekko)
+                basic_machine=powerpc-eabi
+                os=-elf
+                ;;
 	-psos*)
 		os=-psos
 		;;
diff --git a/xmlIO.c b/xmlIO.c
index 8fc00e3..d7e9d20 100644
--- a/xmlIO.c
+++ b/xmlIO.c
@@ -1145,7 +1145,7 @@ xmlGzfileOpen_real (const char *filename) {
     gzFile fd;
 
     if (!strcmp(filename, "-")) {
-        fd = gzdopen(dup(0), "rb");
+        fd = gzdopen(0, "rb");
 	return((void *) fd);
     }
 
diff --git a/xmllint.c b/xmllint.c
index b7af32f..19317d2 100644
--- a/xmllint.c
+++ b/xmllint.c
@@ -3349,11 +3349,11 @@ main(int argc, char **argv) {
 	     i++;
 #ifdef LIBXML_OUTPUT_ENABLED
 	     format = atoi(argv[i]);
-#endif /* LIBXML_OUTPUT_ENABLED */
 	     if (format == 1) {
 	         noblanks++;
 	         xmlKeepBlanksDefault(0);
 	     }
+#endif /* LIBXML_OUTPUT_ENABLED */
 	}
 #ifdef LIBXML_READER_ENABLED
 	else if ((!strcmp(argv[i], "-stream")) ||
